lnwallet/channel.go

```diff
@@ -7326,26 +7326,34 @@ func (lc *LightningChannel) NewAnchorResolutions() (*AnchorResolutions,
 // NewAnchorResolution returns the information that is required to sweep the
 // local anchor.
 func NewAnchorResolution(chanState *channeldb.OpenChannel,
-	commitTx *wire.MsgTx,
-	keyRing *CommitmentKeyRing) (*AnchorResolution, error) {
+	commitTx *wire.MsgTx, keyRing *CommitmentKeyRing,
+	isLocalCommit bool) (*AnchorResolution, error) {
 
 	// Return nil resolution if the channel has no anchors.
 	if !chanState.ChanType.HasAnchors() {
 		return nil, nil
 	}
 
-	// Derive our local anchor script.
-	localAnchor, _, err := CommitScriptAnchors(
+	// Derive our local anchor script. For taproot channels, rather than
+	// use the same multi-sig key for both commitments, the anchor script
+	// will differ depending on if this is our local or remote
+	// commitment.
+	localAnchor, remoteAnchor, err := CommitScriptAnchors(
 		chanState.ChanType, &chanState.LocalChanCfg,
 		&chanState.RemoteChanCfg, keyRing,
 	)
 	if err != nil {
 		return nil, err
 	}
+	if chanState.ChanType.IsTaproot() && !isLocalCommit {
+		localAnchor, remoteAnchor = remoteAnchor, localAnchor
```

Comment:
So the change here is that the keys now differ depending on if this is the remote commit or not, since the local/remove keys are relative, while before it used the multi-sig key which is always absolute.

Reply:
banging my head to understand the relationship here now...isn't the info needed already embedded in the `keyRing` and the correct key will be chosen there? the `isLocal` logic appears in both `CommitScriptAnchors` and here, maybe one of them is redundant?